### PR TITLE
Fix crash on new chat creation due to undefined sequenceNumber

### LINKED ISSUE #2704: Unable to create new group
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
Group should be created and able to initiate chat

## Actual Result:
Blank screen is displayed. Unable to create a group.

## Action Performed:
1. Launch App and login as rybkina+1130@gmail.com
2. Click on Create Group
3. Search for at least 4 users and add them to the group / some of user doesn't have chat with rybkina+1130@gmail.com
4. Click on Create Group

## Workaround:
Need to kill the app and reopen 
## Platform:
Where is this issue occurring?

 Web ✔️
 iOS ✔️
 Android ✔️
 Desktop App ✔️
 Mobile Web ✔️

**Version Number:** 1.0.38-0
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/117210030-28d2d380-adc5-11eb-8583-76d3b4156ad8.mp4

**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

### PR DESCRIPTION
**Pull request by marcaaron**
cc @tgolen @Gonals I think this change is safe but need a second opinion. another option could be to not mark the report as "read" until the `reportMaxSequenceNumber[reportID]` exists, but that's difficult since there's no way to know this has happened because of the async Onyx callbacks etc.

### Details
- When a chat is first created it does not yet have a `reportMaxSequenceNumbers[reportID]`
- We are still trying to mark it as "read" but since it has no history yet it crashes and leads to a white screen of death.

**Bonus:** The rest of the code changes are related to `propTypes` and thought I'd sneak them in here.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/2704

### QA Steps
1. Create a new chat with a user that you have no existing chat with via New Group, New Chat, or Search flows
2. Verify you do not experience a white screen and are brought to the chat.
3. Test "new marker" and unread behavior for regressions

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

File: src/libs/actions/Report.js [hunk 1]
```diff
@@ -940,9 +940,16 @@ function addAction(reportID, text, file) {
  *  is last read (meaning that the entire report history has been read)
  */
 function updateLastReadActionID(reportID, sequenceNumber) {
+    // If we don't have a maxSequenceNumber for this report then we will assume that the report has just been created
+    // and use 0 for the maxSequenceNumber. Otherwise, we'll have an undefined sequenceNumber in cases where no
+    // sequenceNumber is explicitly passed.
+    const maxSequenceNumber = !_.isUndefined(reportMaxSequenceNumbers[reportID])
```
Comment (tgolen) [inline src/libs/actions/Report.js:946]: Should this be an early return instead? Just thinking that if the `maxSequenceNumber` is `0`, then it doesn't really need to call `setLocalLastRead()` and make an API call.

File: src/pages/home/sidebar/OptionRow.js [hunk 2]
```diff
@@ -194,15 +194,15 @@ const OptionRow = ({
                         <View style={[styles.flexRow, styles.alignItemsCenter]}>
                             {option.hasDraftComment && (
                                 <View style={styles.ml2}>
-                                    <Icon src={Pencil} height="16" width="16" />
+                                    <Icon src={Pencil} height={16} width={16} />
```
Comment (tgolen) [inline src/pages/home/sidebar/OptionRow.js:197]: Thanks for fixing this :D It must have been causing a propType warning? I'm ashamed I didn't notice that

**Review by tgolen — CHANGES_REQUESTED** [review]

File: src/pages/home/sidebar/OptionRow.js (hunk 2 above)
Comment (marcaaron) [reply src/pages/home/sidebar/OptionRow.js:197]: Oh yes forgot to mention that I fixed a few random `propTypes` things in this.

File: src/libs/actions/Report.js (hunk 1 above)
Comment (marcaaron) [reply src/libs/actions/Report.js:946]: Oh hmm yeah I guess that would make sense wouldn't it 😄

Comment (marcaaron) [reply src/libs/actions/Report.js:946]: Probably a safer change as well since my comment above could end up being wrong at one point ...

**Comment by marcaaron** [comment]
Updated.

**Review by tgolen — APPROVED** [review]

**Comment by Gonals** [comment]
Ahh. Dammit. I thought this was no longer happening

**Review by Gonals — APPROVED** [review]

**Comment by Gonals** [comment]
All yours, @stitesExpensify!

**Comment by marcaaron** [comment]
Gonna merge this since it's the last deploy blocker I think

**Comment by roryabraham** [comment]
Also verified this on iOS 👍

**Comment by isagoico** [comment]
Android, web and mweb were a pass too 🎉

**Pull request merged**